addons/pos_cash_rounding/static/src/js/pos_cash_rounding.js

```diff
@@ -0,0 +1,97 @@
+odoo.define('pos_cash_rounding.cash_rounding', function (require) {
+    "use strict";
+
+var models = require('point_of_sale.models');
+var rpc = require('web.rpc');
+var screens = require('point_of_sale.screens');
+var utils = require('web.utils');
+
+var round_pr = utils.round_precision;
+
+
+models.load_models([{
+    model: 'account.cash.rounding',
+    fields: ['name', 'rounding', 'rounding_method'],
+    domain: function(self){return [['id', '=', self.config.rounding_method[0]]]; },
+    loaded: function(self, cash_rounding) {
+        self.cash_rounding = cash_rounding;
+    }
+},
+]);
+
+var _super_order = models.Order.prototype;
+models.Order = models.Order.extend({
+    export_for_printing: function() {
+      var result = _super_order.export_for_printing.apply(this,arguments);
+      result.total_rounded = this.get_total_with_tax() + this.get_rounding_applied();
+      result.rounding_applied = this.get_rounding_applied();
+      return result;
+    },
+    get_due: function(paymentline) {
+      var due  = _super_order.get_due.apply(this, arguments);
+      due += this.get_rounding_applied();
+      return round_pr(due, this.pos.currency.rounding);
+    },
+    get_change: function(paymentline) {
+      var change  = _super_order.get_change.apply(this, arguments);
+      if (change !== 0) {
+          change -= this.get_rounding_applied();
+      }
+      return round_pr(change, this.pos.currency.rounding);
+    },
+    get_rounding_applied: function() {
+        if(this.pos.config.cash_rounding) {
+            var total = round_pr(this.get_total_with_tax(), this.pos.cash_rounding[0].rounding);
+
+            var rounding_applied = total - (this.pos.config['iface_tax_included'] === "total"? this.get_subtotal(): this.get_total_with_tax());
```

Comment:
I don't get this condition. We should always round on the total, so why would we need `get_subtotal`?